opentracing/README.md

```diff
@@ -0,0 +1,121 @@
+# DCP RFC
+
+For background:
+- [Main DCP doc](https://paper.dropbox.com/doc/Distributed-Context-Propagation-RGvlvD1NFKYmrJG9vGCES)
+- [DCP "Layers" doc](https://paper.dropbox.com/doc/DCP-Layers-and-Semantics-X1Tm1MSiBJECBkWYQKS2I)
+
+## API overview for those adding instrumentation
+
+Everyday consumers of this `opentracing` package really only need to worry
+about a couple of key abstractions: the `StartSpan` function, the `Span`
+interface, and golang's `context.Context` idiom. Here are code snippets
+demonstrating some important use cases.
+
+A note about `context.Context`: the opentracing API here encourages (but does
+not require) the use of golang-team's `context.Context` scheme. Whether we like
+it or not (this author is on the fence), `context.Context` is here to stay,
+goroutine-local-storage bedamned. When in Rome, do as the Romans do (etc).
+
+#### Singleton initialization
+
+The simplest starting point is `./global.go`. As early as possible, call
+
+    import ".../opentracing"
+    
+    func main() {
+        tracingRecorder := ... // tracing impl specific
+        tracingContextIDSource := ... // tracing impl specific
+        opentracing.InitGlobalTracer(tracingRecorder, tracingContextIDSource)
+        ...
+    }
+
+##### Note: the singletons are optional
+
+If global singletons make you sad, use `opentracing.NewStandardTracer(...)`
+directly and manage ownership of the `opentracing.OpenTracer` explicitly.
+
+#### Creating a Span given an existing Golang `context.Context`
+
+    func xyz(ctx context.Context, ...) {
+        ...
+        sp, ctx := opentracing.StartSpan("span_name", ctx)
```

Comment:
Re the Recorder tags, that may not be the right place to store "service name" or whatever we decide to call it, because from the efficiency considerations I can imagine a recorder to be instantiated once per process/application (especially given the optional "singleton" approach :wink:  ), but an application can expose multiple services.

What if we make `component` and `operation` to be required arguments of _span creation methods_? 

tl;dr;

In Zipkin the service name is encoded in the `Endpoint{host, port, serviceName}` descriptor. Given that an app can listen to multiple ports, in our uber/open-tracing API we opted to take this descriptor in BeginTrace/JoinTrace, as those are the entry points to the service, and then BeginChildSpan inherits the descriptor. If the child span is a local span, it can explicitly provide its `component name`, in the begin-options.

To step back a bit, though, the reason service name is important in Zipkin is because it's the main filter in the UI, and it's also used to build service dependency diagram. If I had a service with 10 different URL routes, I actually wouldn't mind seeing a dependency diagram per service+spanName. That suggests that perhaps service name isn't so special vs. the operation/spanName. I could even ask for aggregation by any other tag on the span, e.g. component+operation+version.